src/Features/LanguageServer/Protocol/RoslynRequestExecutionQueue.cs

```diff
@@ -39,6 +45,55 @@ public override Task WrapStartRequestTaskAsync(Task nonMutatingRequestTask, bool
             }
         }
 
+        /// <inheritdoc/>
+        protected override IMethodHandler GetHandlerForRequest(IQueueItem<RequestContext> work)
+        {
+            var defaultHandlerMetadata = new RequestHandlerMetadata(work.MethodName, work.RequestType, work.ResponseType, LanguageServerConstants.DefaultLanguageName);
+            var defaultHandler = _handlerCache.GetOrAdd(defaultHandlerMetadata, metadata => _handlerProvider.GetMethodHandler(metadata.MethodName, metadata.RequestType, metadata.ResponseType, metadata.Language));
+            var identifier = RoslynRequestExecutionQueue.GetTextDocumentIdentifier(work, defaultHandler);
+            if (identifier is null)
+            {
+                return defaultHandler;
+            }
+
+            var language = _lspWorkspaceManager.GetLanguageForUri(identifier.Uri);
+            if (language is null || language == LanguageServerConstants.DefaultLanguageName)
```

Comment:
GetLanguageForUri should return a non-nullable value.  We should either throw if we can't find the right language, or in the default case return the default name.